homeassistant/components/lcn/device_trigger.py

```diff
@@ -0,0 +1,99 @@
+"""Provides device triggers for LCN."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.automation import (
+    AutomationActionType,
+    AutomationTriggerInfo,
+)
+from homeassistant.components.device_automation import DEVICE_TRIGGER_BASE_SCHEMA
+from homeassistant.components.homeassistant.triggers import event
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_PLATFORM, CONF_TYPE
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.helpers.typing import ConfigType
+
+from . import DOMAIN
+from .const import KEY_ACTIONS, SENDKEYS
+
+TRIGGER_TYPES = {"transmitter", "transponder", "fingerprint", "sendkeys"}
+
+LCN_DEVICE_TRIGGER_BASE_SCHEMA = DEVICE_TRIGGER_BASE_SCHEMA.extend(
+    {vol.Required(CONF_TYPE): vol.In(TRIGGER_TYPES)}
+)
+
+ACCESS_CONTROL_SCHEMA = {vol.Optional("code"): vol.All(vol.Lower, cv.string)}
+TRANSMITTER_SCHEMA = {
+    **ACCESS_CONTROL_SCHEMA,
+    vol.Optional("level"): cv.positive_int,
+    vol.Optional("key"): cv.positive_int,
+    vol.Optional("action"): vol.In([action.lower() for action in KEY_ACTIONS]),
+}
+
+SENDKEYS_SCHEMA = {
+    vol.Optional("key"): vol.In([key.lower() for key in SENDKEYS]),
+    vol.Optional("action"): vol.In([action.lower() for action in KEY_ACTIONS]),
+}
+
+TRIGGER_SCHEMA = vol.Any(
+    LCN_DEVICE_TRIGGER_BASE_SCHEMA.extend(ACCESS_CONTROL_SCHEMA),
+    LCN_DEVICE_TRIGGER_BASE_SCHEMA.extend(TRANSMITTER_SCHEMA),
+    LCN_DEVICE_TRIGGER_BASE_SCHEMA.extend(SENDKEYS_SCHEMA),
+)
+
+
+async def async_get_triggers(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, Any]]:
+    """List device triggers for LCN devices."""
+    device_registry = dr.async_get(hass)
+    device = device_registry.async_get(device_id)
+
+    triggers: list[dict] = []
+    if device.model.startswith(("LCN host", "LCN group", "LCN resource")):  # type: ignore[union-attr]
+        return triggers
+
+    base_trigger = {
+        CONF_PLATFORM: "device",
+        CONF_DOMAIN: DOMAIN,
+        CONF_DEVICE_ID: device_id,
+    }
+
+    triggers.append({**base_trigger, CONF_TYPE: "transmitter"})
+    triggers.append({**base_trigger, CONF_TYPE: "transponder"})
+    triggers.append({**base_trigger, CONF_TYPE: "fingerprint"})
+    triggers.append({**base_trigger, CONF_TYPE: "sendkeys"})
+
+    return triggers
```

Comment:
You could write this as a list comp to avoid the multiple appends
```
TRIGGER_TYPES = ["transmitter", "transponder", "fingerprint", "sendkeys"]
```
```suggestion
    triggers: list[dict] = [{**base_trigger, CONF_TYPE: type_} for type_ in TRIGGER_TYPES]
    return triggers
```

Reply:
Thanks, so obvious. ;-)